packages/swapper/src/swappers/RelaySwapper/utils/relayTokenToAsset.ts

```diff
@@ -0,0 +1,31 @@
+import type { AssetId } from '@shapeshiftoss/caip'
+import type { Asset } from '@shapeshiftoss/types'
+
+import { relayTokenToAssetId } from './relayTokenToAssetId'
+import type { RelayToken } from './types'
+
+export const relayTokenToAsset = (
+  relayToken: RelayToken,
+  assets: Partial<Record<AssetId, Asset>>,
+): Asset => {
+  const assetId = relayTokenToAssetId(relayToken)
+  const maybeAsset = assets[assetId]
+  if (maybeAsset) return maybeAsset
+
+  // It shouldn't happen but...
+  // asset not known by shapeshift
+  // create a placeholder asset using the data we have
```

Comment:
q: Do we want this, and what are the implications of making a swap for an asset we don't support?

Reply:
This is something we already do on Li.Fi side, the implication is that the user won't be able to swap it back to something else through shapeshift except if he adds a custom asset manually (not sure it worth for Solana thought)

Reply:
> But this can't happen, the user has selected the asset in the swapper so shapeshift has the informations, there isn't a world where our swapper present an asset we don't support

As you mentioned, this can't happen at runtime and if it does, we have bigger problems.
Any reason to keep this except for "this is copied from LifiSwapper"?

Reply:
Removed and thrown in [29fc44d](https://github.com/shapeshift/web/pull/9188/commits/29fc44d8ce158086f2cbec2c3c7ee0dd44302dac)